Mage/src/main/java/mage/game/mulligan/MulliganType.java

```diff
@@ -0,0 +1,43 @@
+package mage.game.mulligan;
+
+public enum MulliganType {
+
+    GAME_DEFAULT("Game Default"),
+    VANCOUVER("Vancouver"),
+    PARIS("Paris"),
+    LONDON("London"),
+    CANADIAN_HIGHLANDER("Canadian Highlander");
+
+    private final String displayName;
+
+    MulliganType(String displayName) {
+        this.displayName = displayName;
+    }
+
+    public Mulligan getMulligan(int freeMulligans) {
+        switch(this) {
+            case PARIS:
+                return new ParisMulligan(freeMulligans);
+            case CANADIAN_HIGHLANDER:
+                return new CanadianHighlanderMulligan(freeMulligans);
+            case LONDON:
+                return new LondonMulligan(freeMulligans);
+            default:
+            case VANCOUVER:
```

Comment:
Can there be a case inside a default?

Reply:
The `default` case will fall through to the next option, which is VANCOUVER.  I could add an explicit return of VancouverMulligan inside the default option if you'd like.  Is there a specific concern to this pattern?

Reply:
It is not a concern at all, it is just that I have never seen a dedicated case in a default . I even did not know it was possible. I do believe you it works though 👍 I always write:

case A:
 doThis();
case B:
  doThat();
default:
  doSomethingElse();